src/Connection.php

```diff
@@ -159,7 +165,21 @@ private function receive($len = null)
     {
 
         if ($len) {
-            $line = fread($this->streamSocket, $len);
+            $chunkSize = $this->chunkSize;
+            $line = null;
+            $receivedBytes = 0;
+            while ($receivedBytes < $len) {
+                $bytesLeft = $len - $receivedBytes;
+                if ( $bytesLeft < 1500 ) {
+                    $chunkSize = $bytesLeft;
+                }
+
+                $line .= fread($this->streamSocket, $chunkSize);
+                $receivedBytes += $chunkSize;
+            }
+            if (strlen($line) > 2) {
+                $line = substr($line, 0, -2);
```

Comment:
This line create the regression, basically if there is escape char at the end of message, using rtrim is safer, but I'm not sure if the need to be done here. Maybe you should double check if you don't introduce those two bytes in your own code.